src/librbd/librbd.cc

```diff
@@ -263,12 +263,31 @@ void group_info_cpp_to_c(const librbd::group_info_t &cpp_info,
   c_info->pool = cpp_info.pool;
 }
 
-void group_snap_info_cpp_to_c(const librbd::group_snap_info_t &cpp_info,
+void group_snap_info_cpp_to_c(const librbd::group_snap_info2_t &cpp_info,
 			      rbd_group_snap_info_t *c_info) {
   c_info->name = strdup(cpp_info.name.c_str());
   c_info->state = cpp_info.state;
 }
 
+void group_snap_info2_cpp_to_c(const librbd::group_snap_info2_t &cpp_info,
+                               rbd_group_snap_info2_t *c_info) {
+  c_info->id = strdup(cpp_info.id.c_str());
+  c_info->name = strdup(cpp_info.name.c_str());
+  c_info->image_snap_name = strdup(cpp_info.image_snap_name.c_str());
+  c_info->state = cpp_info.state;
+  c_info->image_snaps_count = cpp_info.image_snaps.size();
+  c_info->image_snaps = static_cast<rbd_group_image_snap_info_t*>(calloc(
```

Comment:
Check for NULL return (calloc failure).

Reply:
The rest of the `src/librbd/librbd.cc` doesn't check for memory allocation errors in these helpers, most likely because it's very hard to get any `malloc()`-like function to fail on Linux.

If we want to be serious to get serious about it, returns from `strdup()`, etc should be checked as well, and that should be done consistently.  It's a topic for another tracker/PR.

Reply:
Yes, all the memory allocation functions should be checked and the functions should return an error if any of them fail. I agree, it can be a separate PR.